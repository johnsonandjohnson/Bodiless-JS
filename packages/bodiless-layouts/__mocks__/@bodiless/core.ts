const editContext = {
  activate: jest.fn(),
  isEdit: true,
};

export const useEditContext = jest.fn(() => editContext);

const activateOnEffect = {
  setId: jest.fn(),
};
export const useActivateOnEffect = () => activateOnEffect;

export const useNode = jest.fn();

const contextMenuFormInner = jest.fn();
export const contextMenuForm = jest.fn(() => contextMenuFormInner);
