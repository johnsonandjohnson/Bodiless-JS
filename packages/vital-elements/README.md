# `@bodiless/vital-elements`

For documentation on the Vital Elements package, please see:

- [Vital Elements | BodilessJS Docs](https://johnsonandjohnson.github.io/Bodiless-JS/#/VitalDesignSystem/Components/VitalElements/)
- [`doc/` Directory](./doc)
