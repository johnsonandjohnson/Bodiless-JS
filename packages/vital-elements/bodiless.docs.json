{
  "VitalDesignSystem": {
    "README.md": "./doc/VitalDesignSystem.md",
    "GettingStarted": {
      "README.md": "./doc/GettingStarted/README.md",
      "CoreConcepts.md": "./doc/GettingStarted/CoreConcepts.md",
      "ComponentsAndTokens.md": "./doc/GettingStarted/ComponentsAndTokens.md",
      "CompositionFromWithout.md": "./doc/GettingStarted/ComposingFromOutside.md",
      "ReachingInside.md": "./doc/GettingStarted/ReachingInside.md",
      "AnatomyOfA_Token.md": "./doc/GettingStarted/AnatomyOfAToken.md",
      "FunctionalClasses.md": "./doc/GettingStarted/FunctionalClasses.md",
      "WorkingWithContent.md": "./doc/GettingStarted/WorkingWithContent.md",
      "CleanComponents.md": "./doc/GettingStarted/CleanComponents.md",
      "YourFirstToken.md": "./doc/GettingStarted/YourFirstToken.md",
      "WhyVital.md": "./doc/GettingStarted/WhyVital.md"
    },
    "Components": {
      "README.md": "./doc/Components.md",
      "VitalElements": {
        "README.md": "./doc/VitalElements.md",
        "ComponentTemplate.md": "./doc/ComponentTemplate.md",
        "ExtendingAndComposingTokens.md": "./doc/ExtendingAndComposing.md",
        "SiteTypography.md": "./doc/SiteTypography.md",
        "Shadow.md": "./doc/Shadow.md",
        "StaticReplacement.md": "./doc/StaticReplacement.md",
        "Tokens": {
          "README.md": "./doc/Tokens.md",
          "TokenDomains.md": "./doc/TokenDomains.md"
        }
      },
      "assets": {
        "PXtoREMTemplate.xlsx": "./doc/assets/PXtoREMTemplate.xlsx"
      }
    }
  }
}
