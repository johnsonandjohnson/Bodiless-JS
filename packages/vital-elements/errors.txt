Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Secondary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Tertiary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Light Theme/Idle
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Form Field/Text/Caption
 . Component color alias "(Typography/Body/Small" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Success
 . Component color alias "(Color/Signal/Success/Success" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Input
 . Component color alias "(Typography/Body/Regular" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Label
 . Component color alias "(Typography/Eyebrow/Eyebrow" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Link
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Hover
 . Component color alias "(Semantic/Color/Interactive/Default" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Active
 . Component color alias "(Semantic/Color/Interactive/Hover" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Bottom
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Top
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Background/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Background/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Accordion/Dropdown/Border/Light Theme/Border
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Light Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Dark Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
