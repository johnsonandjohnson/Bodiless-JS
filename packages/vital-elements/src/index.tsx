/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from './components/AccordionElement';
export * from './components/ArticleCardElement';
export * from './components/BreadcrumbElement';
export * from './components/ButtonElement';
export * from './components/CategoryCardElement';
export * from './components/Color';
export * from './components/DividerElement';
export * from './components/FilterElement';
export * from './components/FloatingButtonElement';
export * from './components/FontSize';
export * from './components/FooterElement';
export * from './components/HeroBannerElement';
export * from './components/ImageCarouselElement';
export * from './components/InputElement';
export * from './components/KnapsackDemoElement';
export * from './components/LinkElement';
export * from './components/ModalElement';
export * from './components/NavigationElement';
export * from './components/PDPProductOverviewElement';
export * from './components/ProductCardElement';
export * from './components/ScrollIndicatorElement';
export * from './components/Spacing';
export * from './components/TextDecoration';
export * from './components/KnapsackDemoElement';
export { default as vitalScrollIndicatorElement } from './generated/vitalScrollIndicatorElement';
export * from './components/Typography';
export * from './util';
