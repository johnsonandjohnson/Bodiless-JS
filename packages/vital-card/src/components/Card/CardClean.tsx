/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { ComponentType } from 'react';
import {
  designable,
  as,
} from '@bodiless/fclasses';
import { withNode } from '@bodiless/core';
import { CardComponents, cardComponentStart, CardBase, asCardToken } from '@bodiless/vital-card-core';
import type { CardProps, CardToken } from '@bodiless/vital-card-core';
import { withoutHydration } from '@bodiless/hydration';

/**
 * This is the base component for cards.
 *
 * @category Component
 *
 * @example
 * **Create a default card:**
 * ```
 * const DefaultCard = on(CardClean)(vitalCardStatic.Default)
 * ```
 * @example
 * **Create a custom card using a token defined in your package**
 * ```
 * const CustomCard = on(CardClean)(myBrandCard.Custom)
 * ```
 */
const CardClean = as(
  designable(cardComponentStart, 'Card'),
  withNode,
)(CardBase);

const CardDescriptionPreview = () => <span className="bl-text-gray-800">Description</span>;

/**
 * Use this version of the card when all components are static.
 *
 * @category Component
 */
const CardStatic: ComponentType<CardProps> = withoutHydration()(CardClean);

export default CardClean;
export {
  CardStatic,
  CardComponents,
  asCardToken,
  CardDescriptionPreview,
};

export type { CardToken };
