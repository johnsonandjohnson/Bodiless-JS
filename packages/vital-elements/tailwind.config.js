/* eslint-disable import/no-dynamic-require, global-require */
/**
 * Copyright Â© 2021 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { getPackageTailwindConfig } from '@bodiless/fclasses';

const tailwindcssDir = require('tailwindcss-dir')();
const designTokens = require('./assets/design-tokens.nested.json');

// Parse the JSON Tokens data
const parsedDesignTokens = JSON.parse(JSON.stringify(designTokens));

/**
 * The `design-tokens.nested.json` file should only contain all Core tokens.
 * These core tokens are then used to extend the corresponding Tailwind config sections.
 *
 * @TODO: The `border.style` tokens do not have a corresponding key in the Tailwind config.
 * `borderStyle` is only a valid key for the TW variants and plugins. The default TW values for
 * borderStyle match the provided tokens.
 *
 * @TODO: `typography` tokens include `text-decoration` and `text-transform` tokens which are
 * not intended to be extended or modified in the Tailwind config and do not have corresponding
 * keys in the Tailwind config. The default TW config covers these tokens and provides the same
 * Default classes like `capitalize` or `uppercase`.
 *
 * @TODO: Also note that currently the `design-tokens.nested.json` is manually adjusted to
 * the more favourable format to resolve certain issues with resolving Figma Plugin aliases.
 * Thehere is a script being worked on to format tokens automatically.
 *
 * @TODO: The `border-width` and `border-style` tokens in JSON are combined under the same key.
 * Tailwind treats it as two separate styles with different config key. In manually updated JSON
 * it is split into two different token "section" like `border.width` and `boder.style`
 * to avoid TW config issues.
 */
const {
  // colors,
  // spacing,
  // typography,
  opacity,
  // border,
} = parsedDesignTokens;

const resolver = (pkgName) => require.resolve(pkgName);

const twConfig = {
  content: [
    './lib/**/!(*.d).{ts,js,jsx,tsx}',
  ],
  theme: {
    fontSize: {
      xs: ['0.75rem', '1.125rem'],
      sm: ['0.875rem', '1.25rem'],
      base: ['1rem', '1.375rem'],
      lg: ['1.75rem', '1.875rem'],
      xl: ['1.625rem', '2rem'],
      '2xl': ['2rem', '2.25rem'],
      '3xl': ['2.563rem', '2.813rem'],
      'm-xs': ['0.688rem', '0.938rem'],
      'm-sm': ['0.75rem', '1rem'],
      'm-base': ['0.875rem', '1.125rem'],
      'm-lg': ['1.125rem', '1.375rem'],
      'm-xl': ['1.438rem', '1.625rem'],
      'm-2xl': ['1.813rem', '2.125rem'],
      'm-3xl': ['2.25rem', '2.5rem'],
    },
    colors: {
      transparent: 'transparent',
      'vital-primary': {
        brand: '#CA081B',
        'card-bg': '#ffffff',
        'page-bg': '#F4F4F4',
        interactive: '#019881',
        'interactive-active': '#000341',
        divider: '#D8D8D8',
        'body-copy': '#63666A',
        'header-copy': '#212121',
        'footer-copy': '#FFFFFF',
        'button-selected': '#015B4D',
        'button-disabled': '#999999',
      },
      'vital-secondary': {
        eyebrow: '#CC0099',
        'footer-bg': '#2B2B33',
        interactive: '#018571',
        table: '#E5E7EB',
        'table-interactive': '#f3f4f6',
        separator: '#666666',
        search: '#888888',
        'button-selected': '#B5E1DA',
        'button-hovered': '#E1F3F0',
        'button-disabled': '#999999',
        'border-button-selected': '#015B4D',
      },
      white: '#FFFFFF',
    },
    extend: {
      /**
       * Vital 2.0 Tokens coming from Figma plugin.
       */
      colors: {
        // ...colors,
        ColorButtonPrimaryTextLightThemeText: '#ffffff',
        ColorInteractiveLightThemeIdle: '#019881',
        ColorInteractiveLightThemeHover: '#017A67',
        ColorInteractiveLightThemeFocus: '#017A67',
        ColorInteractiveLightThemePressed: '#015B4D',
        ColorInteractiveLightThemeDisabled: '#999999',
      },
      fontFamily: {
        DMSans: ['DM Sans', 'sans-serif'],
        // ...typography['font-family'],
        1: 'DM Sans',
        2: 'DM Sans',
      },
      fontWeights: {
        // ...typography.font,
      },
      fontSize: {
        '14px': '14px',
        '16px': '16px',
      },
      lineHeight: {
        // ...typography.leading,
        5: '120%',
        6: '150%'
      },
      textSizes: {
        // ...typography.text,
      },
      spacing: {
        // ...spacing,
        '0px': '0px',
        '1px': '1px',
        '2px': '2px',
        '3px': '3px',
        '4px': '4px',
        '5px': '4px',
        '6px': '4px',
        '7px': '4px',
        '8px': '4px',
        '9px': '4px',
        '10px': '4px',
        '12px': '4px',
        '14px': '4px',
        '16px': '4px',
        '18px': '4px',
        '20px': '4px',
        '24px': '4px',
        '32px': '4px',
        '36px': '4px',
        '40px': '4px',
        '48px': '4px',
        '64px': '4px',
      },
      borderWidth: {
        // ...border.width,
        '1px': '1px',
      },
      borderRadius: {
        // ...border.rounded,
        0: '0px',
        2: '2px',
        4: '4px',
        6: '6px',
        8: '8px',
        10: '10px',
        12: '12px',
        20: '20px',
        150: '150px',
        600: '600px',
        pill: '9999px',
      },
      opacity: {
        // ...opacity,
      },
    },
  },
  plugins: [
    tailwindcssDir,
  ],
};

module.exports = getPackageTailwindConfig({
  twConfig,
  resolver,
});
