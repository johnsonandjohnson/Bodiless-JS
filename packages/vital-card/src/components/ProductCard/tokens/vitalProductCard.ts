/**
 * Copyright Â© 2023 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { withPlaceholder } from '@bodiless/components';
import { EditorPlainClean, vitalEditorPlain } from '@bodiless/vital-editors';
import { ButtonClean, vitalButton } from '@bodiless/vital-button';
import {
  DefaultDomains, vitalSpacing, vitalTypography, vitalProductCardElement,
} from '@bodiless/vital-elements';
import {
  flowHoc, extendMeta, on, Div, P, Img, addProps, TokenCollection, startWith, as,
} from '@bodiless/fclasses';

import { asProductCardToken } from '../ProductCardClean';
import { CardComponents } from '../../Card/CardClean';
import type { CardToken } from '../../Card/CardClean';
import Base from '../../Card/tokens/Base';

/**
  * Default Product Card Token. By default Product Card has `Image`, `Title` and `Link` slots.
  */
const Default = asProductCardToken({
  ...Base,
  Components: {
    ...Base.Components,
    Eyebrow: undefined,
    Description: undefined,
  },
  Content: {
    Title: withPlaceholder('Product Title'),
    CTAText: withPlaceholder('Product Link'),
  },
  Meta: extendMeta(
    flowHoc.meta.term('Sub Type')('Product'),
  ),
  Theme: {
    TitleWrapper: as(
      vitalTypography.HeadlineMedium,
      vitalProductCardElement.TextLightThemeHeadline,
    ),
  },
  Layout: {
    Wrapper: 'w-full flex flex-col',
    ButtonWrapper: as('flex', vitalSpacing.MarginTopSmall),
  },
  Spacing: {
    TitleWrapper: as(vitalSpacing.MarginTopXSmall, vitalSpacing.MarginBottomXSmall),
  },
});

/**
 * Token that adds a Line Clamping to the Product Card Title.
 * Ensures that the Title doesn't span on more than 3 lines.
 *
 * Note: This token is meant to be layered on top of the `Default` token.
 *
 * @TODO: Consider making this more generic. A function that accepts (lines: number) as an
 * argument since line clamping is different based on card placement / page. Will need to tweak
 * types of `TokenCollection` to allow functions that returns token.
 */
const WithTitleLineClamp = asProductCardToken({
  Theme: {
    Title: 'line-clamp-3',
  },
});

/**
 * Token that adds an Eyebrow slot to the Product Card.
 * Adds the `EyebrowWrapper` slot as a `Div` and the `Eyebrow` slot as default Plain Text Editor.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithEyebrow = asProductCardToken({
  Components: {
    EyebrowWrapper: startWith(Div),
    Eyebrow: on(EditorPlainClean)(vitalEditorPlain.Default),
  },
  Content: {
    Eyebrow: withPlaceholder('Eyebrow Text'),
  }
});

/**
 * Token that adds the Description slot to the Product Card.
 * Adds the `DescriptionWrapper` slot as `P` element and the `Description` slot
 * as default Plain Text Editor.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithDescription = asProductCardToken({
  Components: {
    DescriptionWrapper: startWith(P),
    Description: on(EditorPlainClean)(vitalEditorPlain.Default),
  },
  Content: {
    Description: withPlaceholder('Description Text'),
  }
});

/**
 * Token that adds the Primary Vital Button to the Product Card.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithPrimaryButton = asProductCardToken({
  Components: {
    ButtonWrapper: startWith(Div),
    ButtonLink: startWith(ButtonClean),
  },
  Theme: {
    ButtonLink: vitalButton.WithPrimaryStyle,
  },
  Content: {
    ButtonText: withPlaceholder('Button Text'),
  },
});

/**
 * Token that adds the Secondary Vital Button to the Product Card.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithSecondaryButton = asProductCardToken({
  ...WithPrimaryButton,
  Theme: {
    ButtonLink: as(vitalButton.WithSecondaryStyle, 'w-full text-center'),
  },
});

/**
 * Token that adds the Tertiary Vital Button to the Product Card.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithTertiaryButton = asProductCardToken({
  ...WithPrimaryButton,
  Theme: {
    ButtonLink: vitalButton.WithTertiaryStyle,
  },
});

/**
 * A token that adds a placeholder ratings to the Product Card.
 *
 * Note: This token is meant to be layered on top of the `Product` token.
 */
const WithRatings = asProductCardToken({
  Components: {
    RatingWrapper: startWith(Div),
    Rating: on(Img)(addProps({
      src: 'https://svgshare.com/i/sTg.svg',
      alt: 'Stars with 4.1 rating, out of 5 max.',
      title: 'Rating stars',
    })),
  },
  Theme: {
    Rating: 'brightness-50',
  }
});

export interface VitalProductCard extends TokenCollection<CardComponents, DefaultDomains> {
  /**
   * Defines the Product card for the Vital DS.
   * - Extends the Base card with vertical orientation.
   */
  Default: CardToken,
  /**
   * Token that adds an Eyebrow to the Product Card.
   * Adds the `EyebrowWrapper` slot as a `Div` and the `Eyebrow` slot as default Plain Editor.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithEyebrow: CardToken,
  /**
   * Token that adds the Description slot to the Product Card.
   * Adds the `DescriptionWrapper` slot as `P` element and the `Description` slot
   * as default Plain Text Editor.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithDescription: CardToken,
  /**
   * Token that adds the Primary Vital Button to the Product Card.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithPrimaryButton: CardToken,
  /**
   * Token that adds the Secondary Vital Button to the Product Card.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithSecondaryButton: CardToken,
  /**
   * Token that adds the Tertiary Vital Button to the Product Card.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithTertiaryButton: CardToken,
  /**
   * A token that adds a placeholder ratings to the Product Card.
   *
   * Note: This token is meant to be layered on top of the `Product` token.
   */
  WithRatings: CardToken,
  /**
   * Token that adds a Line Clamping to the Product Card Title.
   * Ensures that the Title doesn't span on more than 3 lines.
   *
   * Note: This token is meant to be layered on top of the `Default` token.
   */
  WithTitleLineClamp: CardToken,
}

const vitalProductCard: VitalProductCard = {
  Default,
  WithEyebrow,
  WithDescription,
  WithPrimaryButton,
  WithSecondaryButton,
  WithTertiaryButton,
  WithRatings,
  WithTitleLineClamp,
};

export default vitalProductCard;
