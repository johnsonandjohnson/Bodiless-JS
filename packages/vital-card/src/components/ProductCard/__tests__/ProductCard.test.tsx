import React from 'react';
import { testTokens } from '@bodiless/vital-elements';
import CardClean from '../../Card/CardClean';
import vitalProductCard from '../tokens/vitalProductCard';

// eslint-disable-next-line jest/valid-describe
describe('ProductCard Tokens', testTokens(CardClean, vitalProductCard));
