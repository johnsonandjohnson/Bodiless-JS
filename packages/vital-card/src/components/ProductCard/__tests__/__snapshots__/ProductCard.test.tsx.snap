// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Tokens Matches snapshot for Default 1`] = `
<a
  className="w-full flex flex-col"
  href="#"
>
  <div>
    <div
      data-no-hydrate={true}
      id="f64e1b3016f7c8d0d277ce020d98ee0d"
      style={
        Object {
          "display": "contents",
        }
      }
    >
      <img
        alt=""
        src="test-file-stub"
        title=""
      />
    </div>
  </div>
  <div>
    <div />
    <h3
      className="text-kenvue-neutrals-black text-20.16px lg:text-23.04px leading-5 font-normal font-2 mb-8px md:mb-16px mt-8px md:mt-16px"
    >
      <span
        data-no-hydrate={true}
        id="6e5e36380662b0a36aed30babe228408"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Product Title",
            }
          }
        />
      </span>
    </h3>
    <div
      nodeKey="description"
    />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithBackground 1`] = `
<a
  className="bg-kenvue-neutrals-white"
>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithDescription 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <p>
      <span
        data-no-hydrate={true}
        id="63a9f0ea7bb98050796b649e85481845"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Description Text",
            }
          }
        />
      </span>
    </p>
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithEyebrow 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div>
      <span
        data-no-hydrate={true}
        id="63a9f0ea7bb98050796b649e85481845"
        style={
          Object {
            "display": "contents",
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Eyebrow Text",
            }
          }
        />
      </span>
    </div>
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithHorizontalOrientation 1`] = `
<a
  className="flex-col md:flex-row w-full flex"
>
  <div
    className="py-0 md:py-0 md:w-1/2 flex flex-col"
  >
    <img
      className="w-full"
    />
  </div>
  <div
    className="px-4 md:content-center md:justify-center md:grow md:w-1/2 flex flex-col"
  >
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithPaddings 1`] = `
<a
  className="p-8px md:p-16px"
>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithPrimaryButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div
      className="mt-16px md:mt-20px flex"
    >
      <a
        className="w-full text-center active:bg-kenvue-green-shade-60 focus:bg-kenvue-green-shade-80 hover:bg-kenvue-green-shade-80 bg-kenvue-green-green px-24px py-16px focus:drop-shadow-button focus:outline-none active:drop-shadow-none"
      >
        <span
          className="text-kenvue-neutrals-white"
        />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithRatings 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <img
        alt="Stars with 4.1 rating, out of 5 max."
        className="brightness-50"
        src="https://svgshare.com/i/sTg.svg"
        title="Rating stars"
      />
    </div>
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithSecondButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
    <div
      className="mt-16px md:mt-20px flex"
    >
      <a
        className="w-full text-center focus:drop-shadow-button focus:outline-none active:drop-shadow-none"
      >
        <span
          className="active:text-kenvue-green-shade-60 focus:text-kenvue-green-shade-80 hover:text-kenvue-green-shade-80 text-kenvue-green-green"
        />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithSecondaryButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div
      className="mt-16px md:mt-20px flex"
    >
      <a
        className="w-full text-center active:border-kenvue-green-shade-60 focus:border-kenvue-green-shade-80 hover:border-kenvue-green-shade-80 border-kenvue-green-green px-24px py-16px border-solid border-1px focus:shadow-button focus:outline-none active:shadow-none"
      >
        <span
          className="group-active:text-kenvue-green-shade-60 focus:text-kenvue-green-shade-80 group-hover:text-kenvue-green-shade-80 text-kenvue-green-green"
        />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithTertiaryButton 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div
      className="mt-16px md:mt-20px flex"
    >
      <a
        className="w-full text-center focus:drop-shadow-button focus:outline-none active:drop-shadow-none"
      >
        <span
          className="active:text-kenvue-green-shade-60 focus:text-kenvue-green-shade-80 hover:text-kenvue-green-shade-80 text-kenvue-green-green"
        />
      </a>
    </div>
  </div>
</a>
`;

exports[`ProductCard Tokens Matches snapshot for WithTitleLineClamp 1`] = `
<a>
  <div>
    <img />
  </div>
  <div>
    <div />
    <h3 />
    <div />
    <div>
      <a>
        <span />
      </a>
    </div>
  </div>
</a>
`;
